src/util.ts

```diff
@@ -92,19 +93,25 @@ export const handleHeadPolling = (
         pollStep = 5000
     } = options;
 
-    return xhrHead(uri)
+    return xhrRequest(uri)
         .then((response: any) => {
             if (attempts > maxAttempts) {
                 return Promise.reject(new Error('Export timeout!!!'));
             }
-            return isPollingDone(response.getHeaders()) ?
-                Promise.resolve({ uri }) :
-                delay(pollStep).then(() => {
-                    return handleHeadPolling(xhrHead, uri, isPollingDone, {
+            const responseHeaders = response.getHeaders();
+            if (isPollingDone(responseHeaders)) {
+                if (responseHeaders.status === 200) {
+                    return Promise.resolve({ uri });
+                }
+                return Promise.reject(new ApiResponseError(response.statusText, response, response.getData()));
```

Comment:
I would expect tests for this `handleHeadPolling` as well :) I know its somehow tested view export.test.ts but when export implementation will change this function will not be covered any more

Reply:
I added some tests. Please review it again :)